### PR TITLE
Inline calls when building the trace.

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

File: tests/c/simplecall.newcg.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+//     foo
+//     jit-state: stop-tracing
+//     --- Begin aot ---
+//     ...
+//     func main($arg0: i32, $arg1: ptr) -> i32 {
+//     ...
+//     --- End aot ---
+//     --- Begin jit-pre-opt ---
+//     ...
+//     {{1}}: i8 = Icmp {{2}}, SignedGreater, 1i32
```
Comment (vext01) [inline tests/c/simplecall.newcg.c:18]: Can we use `%{{}}` for local variable placeholders? This prevents constant operands from matching where we expect a variable operand.

Comment (ptersilie) [reply tests/c/simplecall.newcg.c]: [262fb21](https://github.com/ykjit/yk/pull/1082/commits/262fb214025ac8ac85b65d4ecc542e8630345937)

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 2]
```diff
@@ -156,9 +169,11 @@ impl<'a> TraceBuilder<'a> {
 
         // Decide how to translate each AOT instruction based upon its opcode.
         for (inst_idx, inst) in blk.instrs.iter().enumerate() {
+            println!("inst: {}", inst.to_string(self.aot_mod));
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:172]: left-in debug?

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 3]
```diff
@@ -261,6 +277,16 @@ impl<'a> TraceBuilder<'a> {
                 let const_idx = self.jit_mod.const_idx(&jit_const)?;
                 jit_ir::Operand::Const(const_idx)
             }
+            aot_ir::Operand::Arg(arg_idx) => {
+                // The `Arg` operand is an index for the argument of this function. We can use it
+                // to lookup the AOT argument being passed into this function from the call
+                // instruction. We then lookup what JIT instruction the AOT argument maps to and
+                // return this.
+                // Unwrap is safe since an `Arg` means we are currently inlining a function.
+                // FIXME: Is the above correct? What about args in the control point frame?
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:286]: Good question. Those will be trace inputs...

**Comment by vext01** [comment]
Some aspects I'm (we both are) not 100% sure here. Shall we move on and see what happens?

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 3 above)
Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:286]: Yes, but we might have to map them in a similar way to the trace inputs.

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 2 above)
Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/trace_builder.rs]: [72bb808](https://github.com/ykjit/yk/pull/1082/commits/72bb808055c45ac6daf6e62202ceafbf2dd68c2e)

**Comment by ltratt** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**